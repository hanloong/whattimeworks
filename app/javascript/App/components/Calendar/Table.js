import React from "react";
import Row from "./Row";
import ZoneAutoComplete from "../ZoneAutoComplete";
import { TableHeaderColumn } from "material-ui/Table";

const style = {
  th: {
    minWidth: "250px"
  },
  table: {
    fontFamily: "Roboto, sans-serif"
  }
};

const Table = ({ actions, zones, calendar }) => {
  return (
    <table style={style.table}>
      <thead>
        <tr>
          {zones.map((zone, i) => (
            <TableHeaderColumn key={i}>
              <ZoneAutoComplete
                actions={actions}
                index={i}
                deletable={i > 0}
                value={zone}
              />
            </TableHeaderColumn>
          ))}
          <th style={style.th} key="new-zone">
            <ZoneAutoComplete actions={actions} deletable={false} />
          </th>
        </tr>
      </thead>
      <tbody>
        {calendar.map((row, i) => <Row key={i} row={row} />)}
      </tbody>
    </table>
  );
};

export default Table;
