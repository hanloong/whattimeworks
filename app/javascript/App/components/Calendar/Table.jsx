import React from 'react'
import moment from 'moment-timezone'
import Row from './Row'

class Table extends React.Component {
  render() {
    let { zones, step, steps, baseZone } = this.props

    let zonesHeading = zones.map(zone => <th key={ zone }>{ zone }</th>)
    let zonesBody = [...Array(steps).keys()].map(i =>
      <Row
        key={ i }
        zones={ zones }
        time={ moment.tz(baseZone).startOf("day").add((i * step), "minutes")} />
    )

    return <table className="table table-responsive table-bordered table-striped">
      <thead>
        <tr>{ zonesHeading }</tr>
      </thead>
      <tbody>
        { zonesBody }
      </tbody>
    </table>
  }
}

module.exports = Table
