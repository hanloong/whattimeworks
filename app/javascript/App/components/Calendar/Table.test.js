import React from 'react';
import Table from './Table';
import renderer from 'react-test-renderer';
import initialState from '../../constants/state'
import getCalendarState from '../../selectors/calendar'

test('Print out a table', () => {
  let zones = initialState.zones
  let calendar = getCalendarState(initialState)

  const component = renderer.create(
    <Table
      zones={ zones }
      calendar={ calendar } />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
