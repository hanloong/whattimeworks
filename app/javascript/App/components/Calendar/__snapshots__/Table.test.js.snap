// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Print out a table 1`] = `
<table
  className="table table-responsive table-bordered table-striped"
>
  <thead>
    <tr />
  </thead>
  <tbody>
    <tr
      className=""
    >
      <td>
        Today at 12:00 AM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 12:30 AM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 1:00 AM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 1:30 AM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 2:00 AM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 2:30 AM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 3:00 AM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 3:30 AM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 4:00 AM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 4:30 AM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 5:00 AM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 5:30 AM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 6:00 AM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 6:30 AM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 7:00 AM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 7:30 AM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 8:00 AM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 8:30 AM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 9:00 AM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 9:30 AM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 10:00 AM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 10:30 AM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 11:00 AM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 11:30 AM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 12:00 PM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 12:30 PM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 1:00 PM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 1:30 PM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 2:00 PM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 2:30 PM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 3:00 PM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 3:30 PM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 4:00 PM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 4:30 PM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 5:00 PM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 5:30 PM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 6:00 PM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 6:30 PM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 7:00 PM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 7:30 PM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 8:00 PM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 8:30 PM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 9:00 PM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 9:30 PM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 10:00 PM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 10:30 PM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 11:00 PM
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        Today at 11:30 PM
      </td>
    </tr>
  </tbody>
</table>
`;
